pilot/pkg/serviceregistry/external/conversion.go

```diff
@@ -161,6 +159,24 @@ func convertEndpoint(service *model.Service, servicePort *networking.Port,
 	}
 }
 
+// convertWorkloadInstances translates a WorkloadEntry into ServiceInstances. This logic is largely the
+// same as the ServiceEntry convertInstances.
+func convertWorkloadInstances(wle *networking.WorkloadEntry, services []*model.Service, se *networking.ServiceEntry) []*model.ServiceInstance {
+	out := make([]*model.ServiceInstance, 0)
+	for _, service := range services {
+		for _, port := range se.Ports {
+
+			tlsMode := model.DisabledTLSModeLabel
+			if wle.ServiceAccount != "" {
+				tlsMode = model.IstioMutualTLSModeLabel
+			}
```

Comment:
why ` security.istio.io/tlsMode` not respected

Reply:
I have added support for this